resource-and-scopes doc correction (#3458)
AzureAD/microsoft-authentication-library-for-js: lib/msal-browser/docs/resources-and-scopes.md

  msalInstance.acquireTokenSilent(tokenRequest);
 ```

> In the code snippet above, the user will be prompted for consent once they authenticate and receive an **ID Token** and an **Access Token** with scope `User.Read`. Later, if they request an **Access Token** for `User.Read`, they will not be asked for consent again (in other words, they can acquire a token *silently*). On the other hand, the user did not consented to `Mail.Read` at the authentication stage. As such, they will be asked for consent when requesting an **Access Token** for that scope. The token received will contain all the previously consented scopes (for that specific resource), hence the term *incremental consent*.

Reviewer: ```suggestion
In the code snippet above, the user will be prompted for consent once they authenticate and receive an **ID Token** and an **Access Token** with the scope `User.Read`. Later, if they request an **Access Token** for `User.Read`, they will not be asked for consent again (in other words, they can acquire a token *silently*). 

On the other hand, the user did not consent to `Mail.Read` at the authentication stage, therefore, will be asked for consent when requesting an **Access Token** for `Mail.Read` scope. The token received will contain all the previously consented scopes (for that specific resource), hence the term *incremental consent*.
```

  msalInstance.loginPopup(loginRequest);
  // will fail and fallback to an interactive method prompting a consent screen
  msalInstance.acquireTokenSilent(tokenRequest);

>   // will succeed and return an Access Token with scope "api://<myCustomApiClientId>/My.Scope"

Reviewer: Don't we return `openid`, `profile` as default scopes?

Author: @sameerag can't quite remember how to behavior was when this was written, but not anymore (makes sense if they are considered Graph scopes, hence different resource)

Reviewer 2: `openid` and `profile` are OIDC scopes, not Graph scopes. It just so happens that AAD returns `openid` and `profile` with Graph requests, but not for other resources. I don't remember the reason why this is the case.

  msalInstance.loginPopup(loginRequest);

>   // will fail and fallback to an interactive method prompting a consent screen

Reviewer 2: ```suggestion
  // will fail with InteractionRequiredError due to lack of consent for "Mail.Read" scope. You should fallback to an interactive method in this case.
```

  msalInstance.acquireTokenSilent(tokenRequest);
  // will succeed and return an Access Token with scope "api://<myCustomApiClientId>/My.Scope"

>   msalInstance.acquireTokenSilent(anotherTokenRequest);

Reviewer 2: In all requests above where multiple resources are requested in the `scopes` field, we should move any pre-consent scopes to `extraScopesToConsent`. I don't think we should be demonstrating the pattern of requesting multiple resources in the `scopes` field (and it may fail on the /token call).

Author: done!
